backend/metadata_writer/src/metadata_helpers.py

```diff
@@ -32,9 +32,8 @@ def value_to_mlmd_value(value) -> metadata_store_pb2.Value:
 
 
 def connect_to_mlmd() -> metadata_store.MetadataStore:
-    metadata_service_host = os.environ.get("METADATA_GRPC_SERVICE_SERVICE_HOST", "metadata-grpc-service.kubeflow")
-    metadata_service_port = int(os.environ.get("METADATA_GRPC_SERVICE_SERVICE_PORT", 8080))
-    metadata_service_host = "metadata-grpc-service.kubeflow"
+    pod_namespace = get_env_var_with_default("POD_NAMESPACE", "kubeflow")
```

Comment:
Could you check to see if metadawriter runs in the user's namespace in the multi-user mode or in the kubeflow namespace? If it's the former, then the POD_NAMESPACE will be incorrect similar to my comment on driver.